http/codegen/server.go

```diff
@@ -454,8 +454,10 @@ func {{ .HandlerInit }}(
 		ctx = context.WithValue(ctx, goa.MethodKey, {{ printf "%q" .Method.Name }})
 		ctx = context.WithValue(ctx, goa.ServiceKey, {{ printf "%q" .ServiceName }})
 
+		var err error
```

Comment:
Nitpick but I'm wondering if changing the code so that the `err` variable only gets defined that way in the case that there is no payload wouldn't be better so the change would be:
```go
{{- if .Payload.Ref }}
		payload, err := decodeRequest(r)
		...
- {{- end }}
+ {{- else }}
+    var err error
+ {{- end }}
```

Reply:
Thank you for the feedback! In my opinion, trading a the simple declaration `var err error` for a if statement makes it less readable. But I'm fine with both of course, tell me which one you'd like to see in the PR ;)

Reply:
You're right that simply declaring the error variable makes the Goa code easier to read but it also ends up generating Go code that isn't idiomatic:
```go
var err error
payload, err = decodeRequest(r)
```
While certainly not perfect one of the design goals of the Goa code generators has been to generate idiomatic code so I would tend to favor adding the `if`. Thank you!

Reply:
I'll do ;)